logdna/request_types.go

```diff
@@ -0,0 +1,200 @@
+	var prepared interface{}
+	channelRequests := []channelRequest{}
+
+	if listEntries == nil {
+		return nil
+	}
+	for _, entry := range listEntries {
+		e := entry.(map[string]interface{})
+		prepared = nil
+		switch integration {
+		case EMAIL:
+			prepared = emailChannelRequest(e)
+		case PAGERDUTY:
+			prepared = pagerDutyChannelRequest(e)
+		case WEBHOOK:
+			prepared = webHookChannelRequest(e, diags)
+		default:
+			*diags = append(*diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Cannot format integration channel for outbound request",
+				Detail:   fmt.Sprintf("Unrecognized integration: %s", integration),
+			})
+		}
+		if prepared == nil {
+			continue
+		}
+		channelRequests = append(channelRequests, prepared.(channelRequest))
+	}
+	return &channelRequests
+}
+
+func emailChannelRequest(s map[string]interface{}) channelRequest {
+	var emails []string
+	for _, email := range s["emails"].([]interface{}) {
+		emails = append(emails, email.(string))
+	}
+
+	c := channelRequest{
+		Emails:          emails,
+		Immediate:       s["immediate"].(string),
+		Integration:     EMAIL,
+		Operator:        s["operator"].(string),
+		Terminal:        s["terminal"].(string),
+		TriggerInterval: s["triggerinterval"].(string),
+		TriggerLimit:    s["triggerlimit"].(int),
+		Timezone:        s["timezone"].(string),
+	}
+
+	return c
+}
+
+func pagerDutyChannelRequest(s map[string]interface{}) channelRequest {
+	c := channelRequest{
+		Immediate:       s["immediate"].(string),
+		Integration:     PAGERDUTY,
+		Key:             s["key"].(string),
+		Operator:        s["operator"].(string),
+		Terminal:        s["terminal"].(string),
+		TriggerInterval: s["triggerinterval"].(string),
+		TriggerLimit:    s["triggerlimit"].(int),
+	}
+
+	return c
+}
+
+func webHookChannelRequest(s map[string]interface{}, diags *diag.Diagnostics) channelRequest {
+	headersMap := make(map[string]string)
+
+	for k, v := range s["headers"].(map[string]interface{}) {
+		headersMap[k] = v.(string)
+	}
+
+	c := channelRequest{
+		Headers:         headersMap,
+		Immediate:       s["immediate"].(string),
+		Integration:     WEBHOOK,
+		Operator:        s["operator"].(string),
+		Method:          s["method"].(string),
+		TriggerInterval: s["triggerinterval"].(string),
+		TriggerLimit:    s["triggerlimit"].(int),
+		URL:             s["url"].(string),
+		Terminal:        s["terminal"].(string),
+	}
+
+  if bodyTemplate := s["bodytemplate"].(string) ;bodyTemplate != "" {
+    var bt map[string]interface{}
+    // See if the JSON is valid, but don't use the value or it will double encode
+    err := json.Unmarshal([]byte(bodyTemplate), &bt)
+
+    if err == nil {
+      c.BodyTemplate = bt
+    } else {
+      *diags = append(*diags, diag.Diagnostic{
+        Severity: diag.Error,
+        Summary:  "bodytemplate is not a valid JSON string",
+        Detail:   err.Error(),
+      })
+    }
+  }
+
+	return c
```

Comment:
indents are wonky in here. there's others too, if the linter isn't catching it we should work out a fix

Reply:
Formatter *should* fix these as well.  Will re-run then check it.

Reply:
`go fmt ./...` fixed them all.  I must have changed it when my IDE was set to soft tabs.